### PR TITLE
Feature/container listing

### PR DESCRIPTION
**Pull request by motasimvd**
### Proposed Changes
* Added Container listing table to Container list page
* Added Filters to Container listing

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
* New page is located at <your_url>/dotAdmin/#/container-new

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

image

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts [hunk 1]
```diff
@@ -1,22 +1,267 @@
...
+        'Are you sure you want to delete this Container?  (This operation cannot be undone)',
+    'message.container.copy': 'Container copied',
+    'message.container.delete': 'Container archived',
+    'message.container.full_delete': 'Container deleted',
+    'message.container.undelete': 'Container unarchived',
+    'message.container.unpublished': 'Container unpublished',
+    'message.container_list.published': 'Containers published',
+    'containers.fieldName.description': 'Description',
+    'containers.fieldName.lastEdit': 'Last Edit',
+    'containers.fieldName.name': 'Name',
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:121]: language keys that don't exist in https://github.com/dotCMS/core/blob/14a1349d5bc65407427be7d6ca912b98cc35b4f3/dotCMS/src/main/webapp/WEB-INF/messages/Language.properties

needs to be added

File: core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.html [hunk 2]
```diff
@@ -123,6 +123,7 @@
     </td>
     <td class="listing-datatable__action-column" style="width: 5%">
         <dot-action-menu-button
+            *ngIf="actions"
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.html:126]: why ?

Comment (motasimvd) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.html:126]: Angular gets stuck in an error loop, if no actions provided to the table

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html [hunk 3]
```diff
@@ -1,4 +1,81 @@
...
+
+    <dot-listing-data-table
+        #listing
+        [actionHeaderOptions]="actionHeaderOptions"
+        [columns]="tableColumns"
+        url="v1/containers"
+        [actions]="[]"
+        [checkbox]="true"
+        dataKey="inode"
+    >
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html:12]: I think we can set `actions` as an empty array on the `DotListingDataTableComponent` component by default. That way we avoid passing an empty array when there are no `actions`.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts [hunk 4]
```diff
@@ -1,22 +1,267 @@
...
         fixture.detectChanges();
+
+        expect(checkbox.binary).toBeTruthy();
     });
 
-    it('should create', () => {
-        expect(component).toBeTruthy();
+    describe('with data', () => {
+        beforeEach(fakeAsync(() => {
+            spyOn<any>(coreWebService, 'requestView').and.returnValue(
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:240]: Maybe:

```suggestion
            spyOn<CoreWebService>(coreWebService, 'requestView').and.returnValue(
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts [hunk 5]
```diff
@@ -1,22 +1,267 @@
...
+            );
+            fixture.detectChanges();
+            tick(2);
+            fixture.detectChanges();
+            dotListingDataTable = fixture.debugElement.query(
+                By.css('dot-listing-data-table')
+            ).componentInstance;
+            spyOn(dotPushPublishDialogService, 'open');
+
+            spyOn<any>(dialogService, 'open').and.returnValue({
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:254]: Maybe:

```suggestion
            spyOn<CoreWebService>(dialogService, 'open').and.returnValue({
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts [hunk 6]
```diff
@@ -1,22 +1,267 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:1]: Do we need it?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts (hunk 5 above)
Comment (motasimvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:254]: Not letting me use the type DialogService here, throws an error `Argument of type 'string' is not assignable to parameter of type 'never'.`

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts (hunk 6 above)
Comment (motasimvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts:1]: Yes, check the comment above.

**Pull request closed without merging**